lib/validate.js

```diff
@@ -0,0 +1,90 @@
+var allowedParams = {
+	"common": ["type", "products", "currency"],
+	"impression": ["list"],
+	"click": ["list"],
+	"detail": ["list"],
+	"add": [],
+	"remove": [],
+	"purchase": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"refund": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"checkout": ["step", "option"],
+	"checkout_option": ["step", "option"]
+};
+
+var allowedProductParams = {
+	"common": ["id", "name", "brand", "category", "variant", "price"],
+	"impression": ["position"],
+	"click": ["position", "coupon"],
+	"detail": ["position", "coupon"],
+	"add": ["quantity", "coupon"],
+	"remove": ["quantity", "coupon"],
+	"purchase": ["quantity", "coupon"],
+	"refund": ["quantity", "coupon"],
+	"checkout": ["quantity", "coupon"],
+	"checkout_option": ["quantity", "coupon"]
+};
+
+function validateHit(ecData, cb) {
```

Comment:
It's not a `handleError` function. `handleError` means that you explicitly expect someone to do something about it, but really, we don't know what someone might do with the error. Some of us might throw, some of us might log, some of us might swallow silently. Hence why it's a a simple `callback`. In reality, it isn't even a callback - it's an `onError` event emitter, although I do make a "success" callback at the very end. But like I said - this is purely for the purpose of making the switch to `void` (or custom logger) easily, when we get sick of the warnings.